narwhal/worker/src/batch_maker.rs

```diff
@@ -199,9 +246,22 @@ impl BatchMaker {
             .with_label_values(&[self.committee.epoch.to_string().as_str(), reason])
             .observe(size as f64);
 
+        // Note: the code below waits on the WAN to send messages to all others, the store
+        // to write and confirm the write, and then the primary to respond to being notified
+        // of a batch. While all this is going on the other batches are not being created and
```

Comment:
I don't totally understand this comment. The code below just creates a future that waits on the store writes and other primaries' responses, which is awaited in parallel with batch creation and processing if there are enough slots below `MAX_PARALLEL_BATCH`?

Reply:
I now update the gauge in there:
```self.node_metrics.parallel_worker_batches.set(batch_pipeline.len() as i64);```

Reply:
That was an only comment, before the pipelining. Removed!